### PR TITLE
Reduce CI and deployment time by reducing the times NPM install is run

### PR DESCRIPTION
**Pull request by josemigallas**
Saves time and money by:

* **Removing node-sass rebuild** _(_:warning:_This could still fail, needs verification)_
Older versions of node-sass required a rebuild each time node modules were installed. After upgrading to the latest we can get rid of this extra step.

* **Committing flow-type typings folder** _(Needs consensus, please read!)_
In order to work with 3rd libraries and Flow, we need to install typings from `flow-typed`. But instead of installing them on every build it's recommended by the [official wiki](https://github.com/flow-typed/flow-typed/wiki/FAQs#why-do-i-need-to-commit-the-libdefs-that-flow-typed-installs-for-my-project) to commit them in our code base. Typings only affect 2 things: development time and our CI's job `lint`, it does nothing compilation or build-wise. **Let's decide wether we:**
  * commit all typings in our repo (as recommended by flow-typed),
  * or install everty time alongside mode modules,
  * or remove `flow` from our lint and leave it as a thing for development

* **Stop Rake managing npm install in asset_precompile task**
Node modules were being installed **3 times** on each build: 1 in `dependencies_npm` and twice in `assets_precompile`. Rake takes care by node modules using `yarn` by default but since we don't use `yarn` anymore, it does make more sense to manage dependencies ourselves, therefore from now on `npm install` will only be run in `dependencies_npm`.
This speeds up build up to x3.

* **Remove NPM progress bar**
By disabling its progress bar, npm takes aprox. 25% less time to finish

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -333,18 +333,17 @@ jobs:
           name: Install NPM dependencies
           command: |
             npm --version
+            npm set progress false
             npm install
             npm dedupe
       - save_cache:
-          key: v1-npm-{{ arch }}-{{ .Branch }}-{{ checksum "package.json" }}
+          key: node-v1-{{ .Branch }}-{{ checksum "package-lock.json" }}
```
Comment (hallelujah) [inline .circleci/config.yml]: Can't we remove the `{{ .Branch }}` so it is shared by all builds
Or is it dangerous:?

**Comment by hallelujah** [comment]
This is not really reviewable 👎 Too many changes.
Can you just focus on the npm install ? Minimal changes.

The goal would be to install dependencies if they are not already installed.
Like bundler does: `bundle check || bundle install`

**Comment by didierofrivia** [comment]
You are asking about what we should do regarding the flow types and at the same time committing them? And regarding node-sass, it would be better if you see if the issue is actually fixed or not, and put aside speculation about the version update per se...

**Comment by josemigallas** [comment]
@didierofrivia this is a pull request so I am speculating and I am committing everything since this is what I propose. They are just potential changes and therefore to be discussed.

I was hoping for a constructive discussion of how we could speed up our CI but I understand this is too much for the team right now so I'll just dump it.

**Pull request closed without merging**